### ISSUE TITLE
AutoComplete Feature: Ability to add custom values to AutoComplete input while in multiple mode

### ISSUE THREAD

**Issue by poppabear8883** [primefaces/primevue#3424]
### Describe the feature you would like to see added

As the title states, would like the `AutoComplete` component to have the ability to add custom values/chips while in multiple mode.
Right now it only allows you to select from the suggestions in multiple mode.

### Describe alternatives you have considered
An alternative feature could be to add suggestions to the `Chips` component.

### Is your feature request related to a problem?
Its not related to any known problems, however it is a requirement for my project **A.S.A.P**

Basically the issue is:
```
Currently the AutoComplete in multiple mode wont allow for custom values.
Currently Chips don't have autocomplete feature. But allows for custom values.
```

### Additional Notes
I would be willing to pay for this feature if it were to be a priority & we discuss a price. 

I believe this feature doesn't just cater to myself. I'm certain there are plenty of other use cases that many others would find useful!

**Comment by tugcekucukoglu** [comment]
`chip` slot is available for this.

**Comment by poppabear8883** [comment]
> `chip` slot is available for this.

I don't understand how you would implement the feature using a chip slot ? Can you elaborate further ?

**Comment by bahadirsofuoglu** [comment]
Hi, I'm asking to understand more clearly. The problem is customizing the chip component for autocomplete multiple mode? If this is, you can use the chip slot as my friend mentioned. If you have a problem using this slot, we would like to reevaluate it.

**Comment by poppabear8883** [comment]
No not at all, I do not want to customize the chip component.

If you use the AutoComplete component in multiple mode it `forces` you to select a suggestion and refuses to allow you to use your own value ie. by pressing the ENTER key to add your custom value to the selected options.

Just as the Chips component itself allows you to do.

The use case here is that if the user is wanting to put in multiple values and one of the values they want is not in the options array then they should be able to use their own custom value.

Not sure how to be any more clear.

Thanks in advance.

**Comment by patrick-cullen** [comment]
Ran into this problem as well- I'm not sure if there's a way to hook into the keydown event for enter and, if there are no suggestions, add a chip for it (custom input case) while still retaining the existing functionality for keydown event for enter where there are suggestions that are highlighted in the AutoComplete dropdown.

**Comment by patrick-cullen** [comment]
A workaround is to add to the suggestions property whatever the value of your custom query is within the complete method.

**Comment by tugcekucukoglu** [comment]
Issue tracker is used for defects only as part of our commitment to quality and continuous improvement in all areas. Enhancements are collected as valuable community feedback and managed internally so moving this enhancement ticket to our internal project management backlog.
